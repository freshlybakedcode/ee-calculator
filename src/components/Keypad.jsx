import React from 'react';

import styled from 'styled-components';

const KeypadWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 20px 30px 30px 30px;
  border-top: 1px solid black;
`;

const Button = styled.button`
  align-items: center;
  background: ${(props) =>
    props.clear
      ? `linear-gradient(85deg, #faa26c 0%, #f2803a 15%)`
      : `linear-gradient(261deg, #d4d4d4 1%, #fffeff 13%)`};
  border: none;
  border-radius: 5px;
  box-shadow: 2px 2px 3px -1px #111;
  color: ${(props) => (props.clear ? 'white' : '#333')};
  cursor: pointer;
  display: flex;
  font-size: 1em;
  font-weight: bold;
  justify-content: center;
  margin: 10px 5px;
  padding: 5px;
  width: calc(25% - 10px);
`;

const Spacer = styled.div`
  display: flex;
  width: 50%;
`;

export const Keypad = ({ handleKeypadPress }) => {
  return (
    <KeypadWrapper>
      <Button clear data-operator="clear" onClick={handleKeypadPress}>
        C
      </Button>
      <Button clear data-operator="clearEntry" onClick={handleKeypadPress}>
        CE
      </Button>
      <Spacer />

      <Button onClick={handleKeypadPress}>7</Button>
      <Button onClick={handleKeypadPress}>8</Button>
      <Button onClick={handleKeypadPress}>9</Button>
      <Button data-operator="divide" onClick={handleKeypadPress}>
        รท
      </Button>

      <Button onClick={handleKeypadPress}>4</Button>
      <Button onClick={handleKeypadPress}>5</Button>
      <Button onClick={handleKeypadPress}>6</Button>
      <Button data-operator="multiply" onClick={handleKeypadPress}>
        ร
      </Button>

      <Button onClick={handleKeypadPress}>1</Button>
      <Button onClick={handleKeypadPress}>2</Button>
      <Button onClick={handleKeypadPress}>3</Button>
      <Button data-operator="subtract" onClick={handleKeypadPress}>
        -
      </Button>

      <Button onClick={handleKeypadPress}>0</Button>
      <Button data-operator="decimal" onClick={handleKeypadPress}>
        .
      </Button>
      <Button data-operator="equals" onClick={handleKeypadPress}>
        =
      </Button>
      <Button data-operator="add" onClick={handleKeypadPress}>
        +
      </Button>
    </KeypadWrapper>
  );
};
