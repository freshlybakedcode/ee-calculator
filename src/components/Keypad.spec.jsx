import { fireEvent, render } from '@testing-library/react';
import { Keypad } from './Keypad';

const mockHandleKeypadPress = jest.fn();

describe('Keypad', () => {
  it('should render as expected', () => {
    const { container } = render(<Keypad />);
    expect(container).toMatchSnapshot();
  });
  it('should call handleKeyPress() on each button press', () => {
    const { getAllByRole } = render(
      <Keypad handleKeypadPress={mockHandleKeypadPress} />
    );
    const buttons = getAllByRole('button');
    buttons.forEach((button) => fireEvent.click(button));
    expect(mockHandleKeypadPress).toHaveBeenCalledTimes(18);
  });
});
