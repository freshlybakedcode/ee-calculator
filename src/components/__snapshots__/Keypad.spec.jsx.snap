// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render as expected 1`] = `
<div>
  <div
    class="sc-bdnxRM fKtVxz"
  >
    <button
      class="sc-gtsrHT elpfVt"
      data-operator="clear"
    >
      C
    </button>
    <button
      class="sc-gtsrHT elpfVt"
      data-operator="clearEntry"
    >
      CE
    </button>
    <div
      class="sc-dlnjwi eWUZYv"
    />
    <button
      class="sc-gtsrHT htxWcZ"
    >
      7
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      8
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      9
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="divide"
    >
      รท
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      4
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      5
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      6
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="multiply"
    >
      ร
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      1
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      2
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      3
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="subtract"
    >
      -
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
    >
      0
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="decimal"
    >
      .
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="equals"
    >
      =
    </button>
    <button
      class="sc-gtsrHT htxWcZ"
      data-operator="add"
    >
      +
    </button>
  </div>
</div>
`;
